pkg/arvo/app/permission-hook.hoon

```diff
@@ -0,0 +1,282 @@
+  ::
+      %remove
+    =/  owner-access=(unit owner-access)  (~(get by synced) path.act)
+    ?~  owner-access
+      [~ this]
+    ?:  &(=(ship.u.owner-access our.bol) (team:title our.bol src.bol))
+      ::  delete one of our.bol own paths
+      :_  %_  this
+            synced  (~(del by synced) path.act)
+            boned  (~(del by boned) [%permission path.act])
+          ::
+            access-control
+              (~(del by access-control) access-control.u.owner-access)
+          ==
+      %-  zing
+      :~  (pull-wire [%permission path.act])
+          ^-  (list move)
+          %+  turn  (prey:pubsub:userlib [%permission path.act] bol)
+          |=  [=bone *]
+          [bone %quit ~]
+      ==
+    ?.  |(=(ship.u.owner-access src.bol) (team:title our.bol src.bol))
+      :: if neither ship = source or source = us, do nothing
+      [~ this]
+    ::  delete a foreign ship's path
+    :_  %_  this
+          synced  (~(del by synced) path.act)
+          boned  (~(del by boned) [%permission path.act])
+        ==
+    (pull-wire [%permission path.act])
+  ==
+::
+++  peer-permission
+  |=  pax=path
+  ^-  (quip move _this)
+  ?>  ?=([* ^] pax)
+  =/  =owner-access  (~(got by synced) pax)
+  ?>  =(our.bol ship.owner-access)
+  ::  scry permissions to check if subscriber is allowed
+  ?>  (permitted-scry (scot %p src.bol) access-control.owner-access)
+  =/  pem  (permission-scry pax)
+  :_  this
+  [ost.bol %diff %permission-update [%create pax pem]]~
+::
+++  diff-permission-update
+  |=  [wir=wire diff=permission-update]
+  ^-  (quip move _this)
+  ?:  (team:title our.bol src.bol)
+    (handle-local diff)
+  (handle-foreign diff)
+::
+++  handle-local
+  |=  diff=permission-update
+  ^-  (quip move _this)
+  ?-  -.diff
+      %create  [~ this]
+      %add     (change-local-permission [%add path.diff who.diff]) 
+      %remove  (change-local-permission [%remove path.diff who.diff])
+    ::
+      %delete
+    ?.  (~(has by synced) path.diff)
+      [~ this]
+    :_  this(synced (~(del by synced) path.diff))
+    [ost.bol %pull [%permission path.diff] [our.bol %permission-store] ~]~
+  ==
+::
+++  change-local-permission
+  |=  [kind=?(%add %remove) pax=path who=(set ship)]
+  ^-  (quip move _this)
+  :_  this
+  %+  weld
+    ?-  kind
+      %add     (update-subscribers [%permission pax] [%add pax who])
+      %remove  (update-subscribers [%permission pax] [%remove pax who])
+    ==
+  =/  access-paths=(unit (set path))  (~(get by access-control) pax)
+  ::  check if this path changes the access permissions for other paths
+  ?~  access-paths
+    ~
+  (quit-subscriptions kind pax who u.access-paths)
+::
+++  handle-foreign
+  |=  diff=permission-update
+  ^-  (quip move _this)
+  ?-  -.diff
+      %create  (change-foreign-permission path.diff diff)
+      %add     (change-foreign-permission path.diff diff)
+      %remove  (change-foreign-permission path.diff diff)
```

Comment:
Does this not work if you check on `?(%create %add %remove)` all at once? I think the axes of `path.` line up for those cases, so iirc should be fine.

Reply:
It doesn't nest, much to my dismay.

Reply:
@pilfer-pandex plz fix